vendor/github.com/containers/image/v5/docker/docker_image_dest.go

```diff
@@ -452,11 +452,22 @@ func (d *dockerImageDestination) TryReusingBlobWithOptions(ctx context.Context,
 // but may accept a different manifest type, the returned error must be an ManifestTypeRejectedError.
 func (d *dockerImageDestination) PutManifest(ctx context.Context, m []byte, instanceDigest *digest.Digest) error {
 	var refTail string
+	// If d.ref.isUnknownDigest=true, then we push without a tag, so get the
+	// digest that will be used
+	if d.ref.isUnknownDigest {
+		digest, err := manifest.Digest(m)
+		if err != nil {
+			return err
+		}
+		instanceDigest = &digest
+	}
+	fmt.Println("---instanceDigest-----:", instanceDigest)
 	if instanceDigest != nil {
```

Comment:
```go
   refTail = digest.String()
} else if instanceDigest != nil {
```

would do the same thing, without overloading the semantics of `instanceDigest`, and without triggering unnecessary digest matching against a value we have just computed.